src/DataGrid.tsx

```diff
@@ -846,19 +835,27 @@ function DataGrid<R, SR, K extends Key>(
     event.preventDefault();
 
     const ctrlKey = isCtrlKeyHeldDown(event);
-    const nextPosition = getNextSelectedCellPosition({
+    const nextPosition = getNextPosition(key, ctrlKey, shiftKey);
```

Comment:
If `Ctrl + Home` is pressed twice then checkbox looses focus. It is because we re-render the cell but `isCellSelected` flag is not changed